src/mame/drivers/scopus.cpp

```diff
@@ -0,0 +1,233 @@
+   - figure out how/if the ROMs of the two models (150 and 180) can
+     be switched (is that via a dipswitch setting?) The model 180
+     contains a Model 150 ROM as well.
+   - emulate the printer port
+   - document the keyboard layout and emulate it
+
+*********************************************************************/
+
+#include "emu.h"
+#include "cpu/i8085/i8085.h"
+#include "machine/i8251.h"
+#include "machine/i8257.h"
+#include "machine/i8212.h"
+#include "video/i8275.h"
+#include "screen.h"
+#include "bus/rs232/rs232.h"
+#include "machine/clock.h"
+
+class sagitta180_state : public driver_device
+{
+public:
+	sagitta180_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_palette(*this, "palette"),
+		m_crtc(*this, "crtc"),
+		m_dma8257(*this, "dma"),
+		m_maincpu(*this, "maincpu"){ }
+
+	/* devices */
+        required_device<palette_device> m_palette;
+        required_device<i8275_device> m_crtc;
+	required_device<i8257_device> m_dma8257;
+	required_device<cpu_device> m_maincpu;
+
+	DECLARE_DRIVER_INIT(sagitta180);
+	DECLARE_WRITE_LINE_MEMBER(hrq_w);
+	DECLARE_READ8_MEMBER(memory_read_byte);
+        I8275_DRAW_CHARACTER_MEMBER(crtc_display_pixels);
+
+
+	// Character generator
+	const uint8_t *m_chargen;
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+};
+
+void sagitta180_state::machine_start()
+{
+        m_palette->set_pen_color(0, rgb_t(0x00,0x00,0x00)); // black
+        m_palette->set_pen_color(1, rgb_t(0x00,0xa0,0x00)); // normal
+        m_palette->set_pen_color(2, rgb_t(0x00,0xff,0x00)); // highlight
+
+	m_chargen = memregion("chargen")->base();
+}
+
+
+const gfx_layout sagitta180_charlayout =
+{
+        8, 8,             /* 8x16 characters - the last 8 lines are always blank */
+        128,                /* 128 characters */
+        1,              /* 1 bits per pixel */
+        {0},                /* no bitplanes; 1 bit per pixel */
+        {0,1,2,3,4,5,6,7},
+        {0, 8, 2*8, 3*8, 4*8, 5*8, 6*8, 7*8 },
+        8*16                /* space between characters */
+};
+
+static GFXDECODE_START( sagitta180 )
+        GFXDECODE_ENTRY( "chargen", 0x0000, sagitta180_charlayout, 0, 1 )
+GFXDECODE_END
+
+I8275_DRAW_CHARACTER_MEMBER(sagitta180_state::crtc_display_pixels) 
+{ 
+        unsigned i; 
+        const rgb_t *palette = m_palette->palette()->entry_list_raw(); 
+        uint8_t chargen_byte = m_chargen[ (linecount & 7) | ((unsigned)charcode << 3) ]; 
+        uint8_t pixels; 
+ 
+        if (lten) { 
+                pixels = ~0; 
+        } else if (vsp != 0 || (linecount & 8) != 0) { 
+                pixels = 0; 
+        } else { 
+                pixels = chargen_byte;
+        }
+
+        if (rvv) {
+                pixels = ~pixels;
+        }
+
+        for (i = 0; i < 7; i++) {
+                bitmap.pix32(y, x + i) = palette[ (pixels & (1U << (7 - i))) != 0 ];
```

Comment:
You're using `bool` in integer context in the expression in parentheses here.  Also, you've created a three-entry palette, but this will only ever access two entries.  Is there a reason for the third entry?

Reply:
This is code copied and adapted from some other driver. It is the first time I deal with the Intel 8275 so I gotta admit this is a bit of poking and guessing and definitely needs to be reviewed and cleaned up.